vcs/git/repo.go

```diff
@@ -16,6 +16,7 @@ import (
 	"golang.org/x/tools/godoc/vfs"
 	"sourcegraph.com/sourcegraph/go-vcs/vcs"
 	"sourcegraph.com/sourcegraph/go-vcs/vcs/gitcmd"
+	"sourcegraph.com/sourcegraph/go-vcs/vcs/internal"
```

Comment:
I've never seen this done before, but I think it will work as expected. The internal rule is [defined](https://golang.org/s/go14internal) as follows:

> An import of a path containing the element “internal” is disallowed if the importing code is outside the tree rooted at the parent of the “internal” directory.

The import path contains the element "internal", so it should apply.